Kernel/Prekernel/init.cpp

```diff
@@ -92,21 +95,30 @@ static void memmove_virt(void* dest_virt, FlatPtr dest_phys, void* src, size_t n
 
 extern "C" [[noreturn]] void init()
 {
-    if (multiboot_info_ptr->mods_count < 1)
+    u32 initrd_module_start = 0;
+    u32 initrd_module_end = 0;
+    if (multiboot_info_ptr->mods_count > 1)
```

Comment:
Technically, we can support having more than one module being present. Or just use 1 and ignore the rest of them. So it might be a good idea to drop this condition.